feat(Accordion): new component

Haythamasalama opened the pull request:
# Accordion

✨ Introducing a new component for the NuxtLabs UI : Toggleable Accordion Panels! 🎉

📝 This pull request introduces the Disclosure component from the Headless UI library. It enables the display of interactive accordion panels for hiding and revealing content.

![image](https://github.com/nuxtlabs/ui/assets/37311945/b7d4b0bb-5123-4de6-ad72-38fccef0af83)
![image](https://github.com/nuxtlabs/ui/assets/37311945/6407b7b6-a57f-42a9-a343-28eab9695e0e)

🚀 Key features of this component include:

- Customizable labels, leading icons, and content for each panel.
- Options to enable/disable individual panels.
- Control over the initial open/closed state of the panels.
- Ability to change the `color` and `size` of the accordion panels.
- Flexible `icon` customization for opening and closing the panels.
- Support for `disabling` specific panels.
- Option to `open all` panels by default.
- Slots for customizing the `button` and `content items`.

❤️ Please review the changes in this PR and provide feedback.

Resolves #156

benjamincanac commented:
I had a quick look and I would rather call this component `Accordion` :) 

Also, what would you think of using the `UButton` for the triggers? It would allow to handle all the styling and the props like here https://github.com/nuxtlabs/ui/blob/dev/src/runtime/components/overlays/Notification.vue#L83 without configuring everything like the sizes, icons, etc.

Haythamasalama commented:
> I had a quick look and I would rather call this component `Accordion` :)
> 
> Also, what would you think of using the `UButton` for the triggers? It would allow to handle all the styling and the props like here https://github.com/nuxtlabs/ui/blob/dev/src/runtime/components/overlays/Notification.vue#L83 without configuring everything like the sizes, icons, etc.

You're right, I don't like the name `Disclosure` It's nice to be an `accordion`. I will work on these changes.

Haythamasalama commented:
@benjamincanac done

Do you think we need to add `as` props for the content of the accordion?

Haythamasalama commented:
@benjamincanac
 Can we review this to work on new components like `breadcrumbs` ?

benjamincanac commented:
I've taken the liberty to make a few changes to your PR. Let me know what you think!

Haythamasalama commented:
@benjamincanac
Nice changes. But what if we make the default variant 'soft' and add this example of the default 'soft' style to showcase an alternative accordion design?

![image](https://github.com/nuxtlabs/ui/assets/37311945/a9de6a50-10f9-4076-bfd6-20b07c342b6a)

> I just have an issue with the disabled field on an item that doesn't get forwarded to the Button.

I will check it

Haythamasalama commented:
Changing the `as` property from `template` to `div` in `<HDisclosureButton as="div">` makes it work correctly

![image](https://github.com/nuxtlabs/ui/assets/37311945/76073c82-1b70-4f3e-bf5b-a7cff793cdae)

benjamincanac commented:
I fixed it by passing the `disabled` prop to the `DisclosureButton` which works with `as="template"`.

Do you still see an issue?

benjamincanac commented:
I've updated the `default` slot example from your proposal.

Haythamasalama commented:
@benjamincanac 

`disabled` It's working well. I updated the `default` solot example and added the default variant `soft`. I think it looks very nice, rather than `solid`.

benjamincanac commented:
I agree, just pushed some changes too. Let me know 😊

Haythamasalama commented:
> I agree, just pushed some changes too. Let me know 😊
Nice tweak! ❤️

benjamincanac commented:
Let's merge this then!

mathieunicolas commented:
Nice component !
Is it possible to add an option that would let us choose if opening a panel should close the others or not ?
I have a usecase in which that would be helpful.

btw, thanks a lot for this amazing UI module !!! you guys are amazing

Haythamasalama commented:
@mathieunicolas

Yes, it's existied!

* You can open all items by enabling `default-open` props.

Chechout here: https://ui.nuxtlabs.com/elements/accordion#open

* Or open a specific item by passing `defaultOpen` in the items array.

Check it out here: https://ui.nuxtlabs.com/elements/accordion#usage

mathieunicolas commented:
Hi @Haythamasalama thanks for your quick reply !
My question was about the behaviour of the accordion when you click on a panel to open it : it would be nice to have the previously opened one automatically closing, so you always have one panel opened !

For example : 
- all panels are closed
- you click panel 2, it opens
- you click panel 3, it opens, and panel 2 closes

Haythamasalama commented:
@mathieunicolas 
Open issue with this description
